web/src/main/java/se/inera/intyg/webcert/web/service/arende/ArendeDraftServiceImpl.java

```diff
@@ -78,9 +71,16 @@ public ArendeDraft getQuestionDraftById(long id) {
     }
 
     @Override
-    public ArendeDraft create(String certificateId, String subject, String message) {
-        final var arendeDraft = createDraft(certificateId, null, message, subject);
-        return arendeDraftRepository.save(arendeDraft);
+    public ArendeDraft create(String certificateId, String subject, String message, String questionId) {
+        ArendeDraft draft = arendeDraftRepository.findByIntygIdAndQuestionId(certificateId, questionId);
+        if (draft != null) {
+            draft.setText(message);
+            draft.setAmne(subject);
+        } else {
+            draft = createDraft(certificateId, questionId, message, subject);
+        }
+        arendeDraftRepository.save(draft);
+        return draft;
```

Comment:
Return the draft that arendeDraftRepository.save(draft) returns. The returned draft must for example contain the id.